File: TopInterview150/TwoPointers/ContainerWithMostWater/solution.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxVolume = 0;
        int p1 = 0;
        int p2 = height.size()-1;

        while (p1 < p2)
        {
            if (min(height[p1], height[p2]) * (p2-p1) > maxVolume)
            {
                maxVolume = min(height[p1], height[p2]) * (p2-p1);
            }
            if (height[p2] < height[p1])
            {
                p2--;
            }
            else p1++;
        }   
        return maxVolume;
    }
};

File: TopInterview150/TwoPointers/ContainerWithMostWater/solution.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        p1 = 0
        p2 = len(height)-1
        maxVolume = 0
        while p1 < p2:
            currentMax = min(height[p1], height[p2]) * (p2-p1)
            maxVolume = max(currentMax, maxVolume)
            if height[p1] < height[p2]:
                p1+=1
            else:
                p2-=1

        return maxVolume